core/module.go

```diff
@@ -74,6 +83,28 @@ func (*Module) TypeDescription() string {
 	return "A Dagger module."
 }
 
+type ModuleDependency struct {
+	Source dagql.Instance[*ModuleSource] `field:"true" name:"source" doc:"The source for the dependency module."`
+	Name   string                        `field:"true" name:"name" doc:"The name of the dependency module."`
```

Comment:
Glad we have this now, if not just as an escape hatch for name conflicts in some nightmarish hellscape where you depend on two modules with the same name (maybe some kind of transitional phase)

Reply:
Yeah this PR ended up essentially solving most of the problems outlined in https://github.com/dagger/dagger/pull/6478. I actually added an integ test case here for the case of "my module has same name as dependency module" being fixed through this mechanism, but now that you mention it I'll update one of the other ones to also get coverage of "two deps with same name" 👍